[colinhia] iP

colinhia opened the pull request:
# **Hi! It's Julie!!**
> "It's like Jarvis, but better." source: [me](https://en.wikipedia.org/wiki/False_attribution).
## Julie is a personal assistant that helps you track those _pesky_ tasks with:
- Command line interface.
- Astonishingly beautiful UI. (yet to be implemented)
- _Personality._ 😉 
And so much more!
### Supported Commands:
1. `list` displays the list of tasks currently stored.
2. `find` lets you search for specific tasks.
3. `delete` lets you delete tasks.
### Road map
- [x] Implement more sass.
- [ ] Implement more relevant features.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
Level-0

Commit:
Level-1: Echo

Commit:
Level-2: Add, List

Commit:
Level-3: Mark as Done

Commit:
Level-4: ToDos, Events, Deadlines

Commit:
A-TextUiTesting

Commit:
Level-5: Handle Errors

Commit:
Level-6 Delete

Commit:
Branch Level-7.1: Storage

Commit:
Placeholder Data File

Commit:
Branch Level 7.2 - Save and Load

Commit:
Branch Level-8.1: Dates and Times

Commit:
Branch Level 8.2: Formatting alignment and Error Handling

Commit:
A-MoreOOP, A-Packages

Commit:
Merge commit '81a9c537afe8492675eb253968939364d65db091'

Commit:
Merge branch 'branch-A'

Commit:
add JUnit Testing

Commit:
add JUnit 2

Commit:
Merge branch 'branch-A'

Commit:
A-Jar

Commit:
Add Javadocs

JJtan2002 commented on gradle/wrapper/gradle-wrapper.jar:
Should remove binary file from repository?

JJtan2002 commented on src/main/java/julie/command/DeleteCommand.java:
```diff
@@ -0,0 +1,19 @@
+        int x = Integer.parseInt(tokens[1]) - 1;
+        Task t = taskList.get(x);
+        taskList.remove(x);
+        UI.wrappedLinePrint(String.format("Okay, I'll remove this task from the list!\n    %s\n You still have %d tasks left!!", t, taskList.size()));
```
Line is more than 120 characters, try to use line break?

JJtan2002 commented on src/main/java/julie/misc/Parser.java:
```diff
@@ -0,0 +1,46 @@
+    }
+    public static Command parse(String input) throws JulieException {
+        CommandToken cmd = CommandToken.fromString(input);
+        return switch (cmd) {
```
I like this idea to return a switch statement!

JJtan2002 reviewed (commented):
Looks good to me! Just some minor changes that can be made. Good job!

Commit:
add Find functionality

Commit:
edit code to follow Coding Standard

Commit:
Merge branch 'branch-Level-9'

Commit:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/julie/misc/Parser.java

linedoestrolling commented on src/main/java/julie/misc/UI.java:
```diff
@@ -0,0 +1,47 @@
+     */
+    public static <E> void addedPrompt(Task t, List<E> list) {
+        UI.wrappedLinePrint(String.format("Nice! I've added it to the list!\n" +
+                "   %s\n %d tasks left!!" , t, list.size()));
```
Just a minor nitpick, maybe should have no space before the comma?

linedoestrolling commented on src/main/java/julie/task/Deadline.java:
```diff
@@ -0,0 +1,33 @@
+    @Override
+    public String toString() {
+        return String.format("[D] %s (by: %s)",
+                super.toString(),
```
Good following of double indentation for extra long lines!

linedoestrolling commented on src/main/java/julie/misc/Parser.java:
```diff
@@ -0,0 +1,68 @@
+            case TODO -> new TodoCommand(input);
+            case DEADLINE -> new DeadlineCommand(input);
+            case EVENT -> new EventCommand(input);
+            case FIND -> new FindCommand(input);
```
Correct indentation of switch case here even though its in a return statement. Excellent!

linedoestrolling reviewed (commented):
Overall the coding style is consistent and there are little to no errors, only extremely minor ones, which made the code very easy to read.

Commit:
Add checkstyle support

Added checkstyle .xml files from the addressbook, for use by the
checkstyle-idea plugin

Commit:
Migrate to GUI

Commit:
Add auto-resizing windows

Commit:
Add assertions

Assertions are used to verify certain assumptions in the code are functioning. Also found
a bug in the Todo task and rectified it

Commit:
Add assertions

Assertions are used to verify certain assumptions in the code are functioning. Also found
a bug in the Todo task and rectified it

Commit:
Improved code quality

In line with code quality requirements, removed the deprecated ByeCommand, and the corresponding
isExit boolean variable, used in the Chat Bot prior to GUI migration.

Commit:
Improve code quality

In line with code quality requirements, removed the deprecated ByeCommand, and the corresponding
isExit boolean variable, used in the Chat Bot prior to GUI migration.

colinhia committed:
Merge pull request #2 from colinhia/branch-A-Assertions

Branch A Assertions

Commit:
Merge branch 'master' into branch-A-CodeQuality

colinhia committed:
Merge pull request #3 from colinhia/branch-A-CodeQuality

Branch a code quality

Commit:
Add tagging functionality

Added support for single word tagging, with the command format tag x tagString. This update has yet to include
saving of tags and other related functionality.

Commit:
Add tagging functionality

Added support for single word tagging, with the command format tag x tagString. This update has yet to include
saving of tags and other related functionality.

Commit:
Merge branch 'master' of https://github.com/colinhia/ip

Commit:
Update README.md

Commit:
Update Jar name
